### ISSUE TITLE
In Chrome 51 test fails to verify Logitech C920 webcam as capable of 1280x720pixels

### ISSUE THREAD

**Issue by mjgraves** [webrtc/testrtc#210]
**Browsers and versions affected**
Version 51.0.2704.63 m (64-bit)

**Description**
In Chrome 51 on Windows 10 test fails to verify Logitech C920 webcam as capable of 1280x720pixels

**Steps to reproduce**

Using Chrome 51 access https://test.webrtc.org/ 
Set C920 as video device
Run test
**Expected results**
Camera should be reported as capable of 1280x720p

**Actual results**

Camera fails at 720p, reporting an error in constraint

Uploaded test results

https://test.webrtc.org/report/AMIfv963BIPmlo5sPOE9t8sscRdo7hMZuuMKlprKq8_-mQtDR7_R7n8KecQDeZ8OQaEzu4_SMFs-aRlM7HPUGOhCatOGRHSa53-wiGSGXIYgbbdFOvUw3hNEompikbGjXJCVeYjg2aqXLFtM3nplSxmeKZJ97JCxDQ

**Comment by KaptenJansson** [comment]
Thanks for reporting, you seem to have another camera (vMix video) that is selected by default that only supports up to 640x480:
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"cameraName: vMix Video"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"actualVideoWidth: 640"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"actualVideoHeight: 480"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"mandatoryWidth: 640"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"mandatoryHeight: 480"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"encodeSetupTimeMs: 283"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"avgEncodeTimeMs: 4"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"minEncodeTimeMs: 0"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"maxEncodeTimeMs: 8"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"avgInputFps: 29"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"minInputFps: 27"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"maxInputFps: 33"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"avgSentFps: 26"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"minSentFps: 0"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"maxSentFps: 30"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"isMuted: false"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"testedFrames: 95"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"blackFrames: 0"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"info":"frozenFrames: 0"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"success":"Average FPS above threshold"}},
{"ts":1464716889548,"name":"test-run","id":6,"args":{"success":"Captured video using expected resolution."}},
{"ts":1464716889549,"name":"test-run","id":6,"args":{"status":"success"}},
{"ts":1464716889561,"name":"test-run","id":8,"args":{"name":"Check resolution 1280x720","status":"running"}},
{"ts":1464716889573,"name":"test-run","id":8,"args":{"error":"getUserMedia failed with error: ConstraintNotSatisfiedError"}},
{"ts":1464716889578,"name":"test-run","id":8,"args":{"error":"getUserMedia failed with error: ConstraintNotSatisfiedError"}},
{"ts":1464716889579,"name":"test-run","id":8,"args":{"status":"failure"}},
{"ts":1464716889583,"name":"test-run","id":9,"args":{"name":"Check supported resolutions","status":"running"}},
{"ts":1464716889586,"name":"test-run","id":9,"args":{"success":"Supported: 160x120"}},
{"ts":1464716889589,"name":"test-run","id":9,"args":{"success":"Supported: 160x120"}},
{"ts":1464716889591,"name":"test-run","id":9,"args":{"success":"Supported: 320x180"}},
{"ts":1464716889592,"name":"test-run","id":9,"args":{"success":"Supported: 320x240"}},
{"ts":1464716889594,"name":"test-run","id":9,"args":{"success":"Supported: 640x360"}},
{"ts":1464716889595,"name":"test-run","id":9,"args":{"success":"Supported: 640x480"}},
{"ts":1464716889597,"name":"test-run","id":9,"args":{"info":"768x576 not supported"}},
{"ts":1464716889598,"name":"test-run","id":9,"args":{"info":"1024x576 not supported"}},
{"ts":1464716889599,"name":"test-run","id":9,"args":{"info":"1280x720 not supported"}},
{"ts":1464716889601,"name":"test-run","id":9,"args":{"info":"1280x768 not supported"}},
{"ts":1464716889602,"name":"test-run","id":9,"args":{"info":"1280x800 not supported"}},
{"ts":1464716889603,"name":"test-run","id":9,"args":{"info":"1920x1080 not supported"}},
{"ts":1464716889605,"name":"test-run","id":9,"args":{"info":"1920x1200 not supported"}},
{"ts":1464716889606,"name":"test-run","id":9,"args":{"info":"3840x2160 not supported"}},
{"ts":1464716889608,"name":"test-run","id":9,"args":{"info":"4096x2160 not supported"}},

Could you try to select the Logitech 920 camera in the settings menu and retry?

Thanks!

**Comment by mjgraves** [comment]
Most curious. I had selected the Logitech C920 camera in that past event. There are a number of real & virtual video sources on this computer. It must have been confused about which device was selected.
 
--------- Original Message --------- Subject: Re: [webrtc/testrtc] In Chrome 51 test fails to verify Logitech C920 webcam as capable of 1280x720pixels (#210)

**Comment by KaptenJansson** [comment]
Had you selected it in the same session (i.e no page reload)? Could you confirm it by selecting in the settings menu, run the tests, then run the tests again without reloading the page and see if the correct camera is used? You can verify this by expanding the check resolution tests as it lists the camera name.

**Comment by mjgraves** [comment]
I tried the test you described. However, I did so with two different USB webcams attached. A C920 and a C930e.
 
I selected the C920 as the camera and run the test, which was successful. As in the past it reported my network as not being ipv6 enabled...which is true.
 
I run the test again, no reloading of the page. This time the C920 camera was correctly reported, but it reported that net network was ipv6 capable!
 
I then selected the C930e camera. run the test a third time. No reload of the page. It persisted in using the C920 camera. Except when it came to the video bandwidth test. I could see that it accessed the C930e camera. The activity light on that camera was lit.
 
I then reloaded the page. The sources were set to defaults. I once again selected the C930e. Run the test and the camera tests involved the C920. The video throughput test invoked the C930e.
 
Something is amiss with device selection when there are multiple devices attached.
 
Michael Graves
 mgraves@mstvp.com
http://www.mgraves.org
o(713) 861-4005
 c(713) 201-1262
 sip:mgraves@mjg.onsip.com
 skype mjgraves

 
--------- Original Message --------- Subject: Re: [webrtc/testrtc] In Chrome 51 test fails to verify Logitech C920 webcam as capable of 1280x720pixels (#210)
